core/lib/libtomcrypt/src/tee_ltc_provider.c

```diff
@@ -1203,13 +1206,19 @@ static TEE_Result rsassa_verify(uint32_t algo, struct rsa_public_key *key,
 		goto err;
 	}
 
+	if (ltc_rsa_algo == LTC_PKCS_1_PSS && salt_len == -1)
+		salt_len = hash_size;
+
 	ltc_res = rsa_verify_hash_ex(sig, sig_len, msg, msg_len, ltc_rsa_algo,
 				     ltc_hashindex, salt_len, &stat, &ltc_key);
-	if ((ltc_res != CRYPT_OK) || (stat != 1)) {
-		res = TEE_ERROR_SIGNATURE_INVALID;
-		goto err;
-	}
-	res = TEE_SUCCESS;
+
+	if (ltc_res == CRYPT_OK)
+		if (ltc_stat == 1)
+			res = TEE_SUCCESS;
+		else
+			res = TEE_ERROR_SIGNATURE_INVALID;
+	else
+		res = TEE_ERROR_GENERIC;
```

Comment:
@jforissier: i agree with out, the test should be modified.
This test is related to the Bleichenbacher attack. It seems that the malicious forged signature to verify is badly formatted from LTC PKCS1 view. Hence LTC returns a format error, not an invalid signature status. On such error OPTEE panics the TA: hence, there is no way to run other Bleichenbacher attack test unless the TA is loaded back.
In few words, the xtest 4011 should be deeper re-implemented than just changing the expected return status ID.

For info: i have tested the GP test suite. At least, it does not complain with this change.